lib/deePool.src.js

```diff
@@ -29,25 +29,25 @@
 		// ******************************
 
 		function use() {
-      if (freePoolEnd == -1) {
-        if (freePool.length == 0) {
-          grow(5);
-        } else {
-          grow();
-        }
-      }
-      return freePool[freePool.length - (freePoolEnd--)];
+			if (freePoolEnd == -1) {
+				if (freePool.length == 0) {
+					grow(5);
+				} else {
+					grow();
+				}
+			}
+			return freePool[freePool.length - 1 - (freePoolEnd--)];
```

Comment:
Yeah, so that's just so when we `grow` the pool we can push onto the end of the backing array instead of unshifting to the front of it. Testing showed major slowdowns with growing the array from the front instead of the end (which makes sense because the vm probably can't take advantage of memcopy efficiently under the hood).